docGen/chainbench.go

```diff
@@ -0,0 +1,183 @@
+	versionedConf := make(map[string]ChainBenchSectionsConfig, 3)
+
+	for _, sectionConfig := range sectionsConfigs {
+		configTypeMap := sectionConfig
+		configTypeMap.Sections = make(map[string]ChainBenchRulesConfig)
+		versionedConf[strconv.FormatInt(semver.MustParse(sectionConfig.ID).Major(), 10)] = configTypeMap
+	}
+
+	for _, ruleConfig := range rulesConfigs {
+		version := strconv.FormatInt(semver.MustParse(ruleConfig.ID).Major(), 10)
+		versionedConf[version].Sections[ruleConfig.ID] = ruleConfig
+	}
+
+	versioned := make(map[string]map[string]ChainBenchSectionsConfig)
+	versioned["cis-1.0"] = versionedConf
+	if err := writeSupplyChainTemplates(versioned, outputDir); err != nil {
+		fmt.Println(err)
+	}
+}
+
+func writeSupplyChainTemplates(versionedConfigs map[string]map[string]ChainBenchSectionsConfig, outputDir string) error {
+	complianceMenu.AddNode("softwaresupplychain", "Software Supply Chain", outputDir, "compliance", []string{},
+		[]menu.BreadCrumb{{Name: "Compliance", Url: "/compliance"}}, "softwaresupplychain", true)
+
+	outputDir = filepath.Join(outputDir, "softwaresupplychain")
+
+	t := template.Must(template.New("bodyContent").Parse(chainBenchTemplate))
+	for version, grouping := range versionedConfigs {
+		complianceMenu.AddNode(version, cisVersion(version), filepath.Join(outputDir),
+			"softwaresupplychain", []string{},
+			[]menu.BreadCrumb{{Name: "Compliance", Url: "/compliance"},
+				{Name: "Software Supply Chain", Url: "/compliance/softwaresupplychain"}}, "softwaresupplychain", true)
+
+		for group, config := range grouping {
+
+			complianceMenu.AddNode(fmt.Sprintf("%s-%s", version, config.Type), config.Text, filepath.Join(outputDir, version),
+				version, []string{},
+				[]menu.BreadCrumb{
+					{Name: "Compliance", Url: "/compliance"},
+					{Name: "Software Supply Chain", Url: "/compliance/softwaresupplychain"},
+					{Name: cisVersion(version), Url: fmt.Sprintf("/compliance/softwaresupplychain/%s", version)},
+				}, "aqua", false)
+
+			for id, checkGroup := range config.Sections {
+
+				targetFilePath := filepath.Join(outputDir, version, fmt.Sprintf("%s-%s", version, config.Type),
+					fmt.Sprintf("%s.md", checkGroup.ID))
+				if err := os.MkdirAll(filepath.Dir(targetFilePath), os.ModePerm); err != nil {
+					return err
+				}
+				var documentBody bytes.Buffer
+				if err := t.Execute(&documentBody, map[string]interface{}{
+					"ShortName":   checkGroup.Name,
+					"ID":          id,
+					"Version":     version,
+					"NiceVersion": cisVersion(version),
+					"Category":    config.Type,
+					"Checks":      checkGroup.Checks,
+					"ParentID":    group,
+					"ParentTitle": config.Text,
+				}); err != nil {
+					return err
+				}
+
+				if err := os.WriteFile(targetFilePath, documentBody.Bytes(), os.ModePerm); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+const chainBenchTemplate string = `---
+title: {{.ShortName}}
+id: {{ .ID }}
+source: Chain Bench
+icon: softwaresupplychain
```

Comment:
You would need to add an icon to themes/aquablank/static/images along the lines of `icon_page_softwaresupplychain.png` or `icon_tile_softwaresupplychain.png` and add to the css - themes/aquablank/static/sass/settings/_avd_hero.scss around line 34

Reply:
I added the Icon and the desired changes.